### PR TITLE
Add jump_to_track method

### PR DESCRIPTION
**Pull request by ghcs27**
As proposed in #513. I will gladly make any change that is requested.

### CODE REVIEW

**Comment by ghcs27** [comment]
The Travis CI build errors on Python 2.6 with
`ERROR: You need Python 2.7 or 3.3+ to install the typing package.`

**Comment by DPH** [comment]
Hi @ghcs27,
tested change and it all works as expected.

In the documentation:  it raises a 701 error if not already playing from the queue, and a 711 error if index out of range. Need to correct this.

I think it would be worth adding a bit of context:
'''This function  can only be used if the player is already set to use the queue. If not use play_from_queue. This is faster than play from queue if already using the queue, as it only make one call rather than 3. Note: if the queue is already playing it will continue to play using the new track after jump_to_track. If paused it will remain paused.
'''
Lets get input from others too
Cheers David

**Comment by ghcs27** [comment]
Thanks for the doc infos! Updated now.

**Comment by KennethNielsen** [comment]
Since @DPH already reviewed this, I will just give this a quick glance. I am annoyed about the Python 2.6 issue, since I prefer not to pull stuff that doesn't pass test. I will check where we are on deprecating Python 2.6.

**Comment by KennethNielsen** [comment]
I think we are moving on removing 2.6 now (#526), so I would ask you to rebase it afterwards

**Comment by KennethNielsen** [comment]
@ghcs27 #526 was merged. Can you rebase against master, so that we can see if the tests pass?

**Comment by ghcs27** [comment]
Done.

File: soco/core.py [hunk 1]
```diff
@@ -432,20 +432,43 @@ def ramp_to_volume(self, volume, ramp_type='SLEEP_TIMER_RAMP_TYPE'):
         return int(response['RampTime'])
 
     @only_on_master
-    def play_from_queue(self, index, start=True):
-        """ Play a track from the queue by index. The index number is
-        required as an argument, where the first index is 0.
+    def jump_to_track(self, index):
+        """ Jump to another track in the queue.
```
Comment (KennethNielsen) [inline soco/core.py]: No space after """ and include a blank line after the first single line and the larger explanation below

File: soco/core.py [hunk 2]
```diff
@@ -432,20 +432,43 @@ def ramp_to_volume(self, volume, ramp_type='SLEEP_TIMER_RAMP_TYPE'):
...
+            If the queue is already playing it will continue to play using the
+            new track after jump_to_track. If paused it will remain paused.
 
-        Returns:
-        True if the Sonos speaker successfully started playing the track.
-        False if the track did not start (this may be because it was not
-        requested to start because "start=False")
+        Arguments:
+            index (int): The index of the track to play; first item in the
+                queue is 0.
```
Comment (KennethNielsen) [inline soco/core.py]: Usually no full stop if there is only one sentence

File: soco/core.py [hunk 3]
```diff
@@ -432,20 +432,43 @@ def ramp_to_volume(self, volume, ramp_type='SLEEP_TIMER_RAMP_TYPE'):
...
 
-        Raises SoCoException (or a subclass) upon errors.
+        Raises:
+            SoCoUPnPException: UPnP Error 701 if the queue isn't currently
+                playing.
+                UPnP Error 711 if the index is invalid.
```
Comment (KennethNielsen) [inline soco/core.py]: While I think it is a good idea to separate the error numbers like this, have you checked how it looks when the docs compile. Sphinx sometimes does weird stuff.

File: soco/core.py [hunk 4]
```diff
@@ -432,20 +432,43 @@ def ramp_to_volume(self, volume, ramp_type='SLEEP_TIMER_RAMP_TYPE'):
...
+            ('Target', index + 1)
+        ])
 
+    @only_on_master
+    def play_from_queue(self, index, start=True):
+        """ Play a track from the queue by index.
+
+        Arguments:
+            index (int): the index of the track to play; first item in the
+                queue is 0.
```
Comment (KennethNielsen) [inline soco/core.py]: As above

File: soco/core.py [hunk 5]
```diff
@@ -432,20 +432,43 @@ def ramp_to_volume(self, volume, ramp_type='SLEEP_TIMER_RAMP_TYPE'):
...
+            start (bool): If the item that has been set should start playing
+
+        Raises:
+             SoCoUPnPException: UPnP Error 711 if the index is invalid.
```
Comment (KennethNielsen) [inline soco/core.py]: and here

**Comment by ghcs27** [comment]
Sure! But can you please tell me how I can check how my docstrings will look in Sphinx?

**Comment by ghcs27** [comment]
Sure! But can you please tell me how I can check how my docstrings will look in Sphinx?

**Comment by KennethNielsen** [comment]
@ghcs27 of course. You will need to compile the docs locally, so you will need to install sphinx locally. If you develop SoCo in a virtual environment this should be fairly easy. If you install the development requirements:
```bash
pip install -r requirements-dev.txt
```

then enter the doc folder, and then do:
```bash
make html
```
then the docs will be built inside the `_build/html` sub-folder and so you should be able to open `_build/html/index.html` in a browser to view the results.

If you are on Windows you will need to use the make.bat. I don't know the syntax for it, but I suspect it is similar and easily googlable.

Let me know if it doesn't work for you.

**Comment by ghcs27** [comment]
@KennethNielsen thanks for the instructions!

I took care of the docstring (I hope it's ok now).

But before you merge anything, please await another PR for comparison which integrates this functionality into seek(). This will (a) be closer to the implementation by Sonos (they both use the avTransport.Seek() method) and (b) allow you to construct a dict `{track=0, position='0:00:00'}` representing the absolute position and seek to this position with a single call (useful for snapshot et al.).

**Comment by KennethNielsen** [comment]
Closed in favor of #541.

**Pull request closed without merging**